src/unix/apple/macosx/app.m

```diff
@@ -1236,6 +1236,33 @@ void MTY_AppSetRelativeMouse(MTY_App *ctx, bool relative)
 	app_apply_relative(app);
 }
 
+// Extracts the Parsec client's private scale info (if any) from the given PNG cursor image.
+// Assumes the scale chunk is the last chunk in the PNG file. 
+double app_extract_private_scale_tag_from_png_cursor(const void *png_ptr, size_t png_len)
+{
+	if (png_len < sizeof(uint32_t) * 4 || !png_ptr)
+		return 1.0;
+
+	const char *tmp = &((const char *) png_ptr)[png_len - sizeof(uint32_t) * 4];
+	const uint32_t *chunk_ptr = (const uint32_t *) tmp;
+
+	#define SWAP32(x) ( (((x) & 0xFF) << 24) | (((x) & 0xFF00) << 8) | (((x) & 0xFF0000) >> 8) | (((x) & 0xFF000000) >> 24) )
+
+	if (SWAP32(chunk_ptr[0]) != sizeof(uint32_t))
```

Comment:
There is a matoya function called `MTY_Swap32` but this is also wrong, because the data in the PNG is always in network byte order, so, to ensure that this code will always work, it should be `MTY_SwapFromBE32` and it will work on all platforms and CPU's.

Reply:
Yes, this does indeed need to be in network byte order.  Thanks for noticing that.  Fixed